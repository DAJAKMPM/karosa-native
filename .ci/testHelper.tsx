import React, { JSXElementConstructor } from "react";
import { View } from "react-native";
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from "@testing-library/react-native";
import { Formik, FormikValues } from "formik";

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };

  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || "wrapper");
  return { wrapper, ...renderApi };
};

export const renderWithFormikWrapper = <T extends FormikValues>(
  initialValues: T,
  children: React.ReactElement<any, string | JSXElementConstructor<any>>
) => {
  return renderWithWrapper(
    <Formik
      initialValues={initialValues}
      onSubmit={() => console.log("submit")}
      component={() => children}
    />
  );
};
