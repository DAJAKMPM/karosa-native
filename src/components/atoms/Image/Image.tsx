/**
 *
 * Image
 * @format
 *
 */

import React, { FC } from "react";
import { Image as RnImage } from "react-native-elements";

import type { PropsType } from "./types";
import ImageStyles from "./styles";

const Image: FC<PropsType> = (props) => {
  const { source, imageStyle, resizeMode, onLoadEnd } = props;

  return (
    <RnImage
      style={[ImageStyles.container, imageStyle]}
      source={source}
      resizeMode={resizeMode}
      onLoadEnd={onLoadEnd}
    />
  );
};

export default Image;
