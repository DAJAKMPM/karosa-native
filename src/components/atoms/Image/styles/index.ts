/**
 *
 * @format
 *
 */

import { StyleSheet } from "react-native";

const ImageStyles = StyleSheet.create({
  container: {
    width: 50,
    height: 50,
  },
});

export default ImageStyles;
