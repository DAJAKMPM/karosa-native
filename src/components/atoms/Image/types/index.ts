/**
 *
 * @format
 *
 */

import {
  ImageResizeMode,
  ImageSourcePropType,
  ImageStyle,
  StyleProp,
} from "react-native";

export type PropsType = {
  source: ImageSourcePropType;
  resizeMode?: ImageResizeMode;
  customStyle?: StyleProp<ImageStyle>;
};
