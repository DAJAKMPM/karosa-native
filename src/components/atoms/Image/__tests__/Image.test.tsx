import React from "react";
import { renderWithWrapper } from "@app/ci/testHelper";
import Image from "../Image";

const FAKE_URI = "https://i.imgur.com/0y8Ftya.jpg";

describe("Image Component", () => {
  it("should match snapshot", () => {
    const { toJSON } = renderWithWrapper(
      <Image
        source={{
          uri: FAKE_URI,
        }}
      />
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
