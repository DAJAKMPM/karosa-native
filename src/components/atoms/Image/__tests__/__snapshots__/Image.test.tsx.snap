// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <Image
      onLoad={[Function]}
      source={
        Object {
          "uri": "https://i.imgur.com/0y8Ftya.jpg",
        }
      }
      style={
        Object {
          "bottom": 0,
          "height": 50,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 50,
        }
      }
      testID="RNE__Image"
      theme={
        Object {
          "colors": Object {
            "black": "#242424",
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "grey": "rgba(0, 0, 0, 0.54)",
                "primary": "#2196f3",
                "searchBg": "#dcdce1",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "default": Object {
                "error": "#ff3b30",
                "grey": "#7d7d7d",
                "primary": "#007aff",
                "searchBg": "#dcdce1",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
              "ios": Object {
                "error": "#ff3b30",
                "grey": "#7d7d7d",
                "primary": "#007aff",
                "searchBg": "#dcdce1",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
              "web": Object {
                "error": "#ff190c",
                "grey": "#393e42",
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#ca71eb",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#ca71eb",
            "success": "#52c41a",
            "warning": "#faad14",
            "white": "#ffffff",
          },
        }
      }
      transition={true}
      transitionDuration={360}
    />
    <View
      accessibilityElementsHidden={true}
      importantForAccessibility="no-hide-descendants"
      pointerEvents="none"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "opacity": 1,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#bdbdbd",
            "height": 50,
            "justifyContent": "center",
            "width": 50,
          }
        }
        testID="RNE__Image__placeholder"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <ActivityIndicator
            animating={true}
            color="#1C1C1C"
            hidesWhenStopped={true}
            size="small"
          />
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "height": 50,
            "width": 50,
          },
          undefined,
        ]
      }
      testID="RNE__Image__children__container"
    />
  </View>
</View>
`;
